14.0 - [IMP] fieldservice_equipment_stock

marcos-mendez opened the pull request:
This feature allow only real PRODUCT EQUIPMENT to be indexed in the Maintenance Equipment form FSM.EQUIPMENT VIEW

marcos-mendez committed:
Update fsm_equipment.xml

This feature allow only real PRODUCT EQUIPMENT to be indexed in the Maintenance Equipment form FSM.EQUIPMENT VIEW

max3903 commented:
@marcos-mendez `fieldservice_equipment_stock` does not depend on `maintenance` and should not depend on it.

marcos-mendez commented:
@max3903 it is to allow only equipments as it has nothing to DO on Service product, Service products and consumable products shouldnt be shown in this views. 

You can test it here https://github.com/kimenz/field-service/pull/1

max3903 commented:
@marcos-mendez It is working on https://github.com/kimenz/field-service/pull/1 because of `('type', '=', 'product')`.

Your change is using the field `maintenance_ok`. This field does not exist if `maintenance` is not installed.

marcos-mendez commented:
Yes @max3903 thats right because we were having the option to put consumable toilet paper in te fieldsetrvice equipment it was ultrageous. My boss called me and gave me a scolding. I told him i can't be responsible by the user in a Homer Simpson Style. Sorry i don't know the other stuff, tests and pre-commit trying to make my best here

max3903 commented:
@marcos-mendez Correction: This field does not exist even if `maintenance` is installed. Which module is adding this field?

marcos-mendez commented:
In portugues this field is show as is FSM equipment

max3903 commented on fieldservice_equipment_stock/views/fsm_equipment.xml:
```diff
@@ -8,7 +8,10 @@
-                    <field name="product_id" />
+                    <field 
+                        name="product_id"
+                        domain="[('maintenance_ok', '=', true)]"
```
```suggestion
                        domain="[('type', '=', "product")]"
```

marcos-mendez commented:
maintenance_ok is not the field is product is fsm.equipment here. Sorry was a mess my PR

max3903 commented:
@marcos-mendez Check my comments on the code. I believe that is what you are looking for.

marcos-mendez commented:
Really Sorry max i dont know that much git. I'll try to make my best

max3903 commented:
@marcos-mendez Don't be sorry. Take your time to read the Git documentation...

max3903 commented on fieldservice_equipment_stock/views/fsm_equipment.xml:
```diff
@@ -8,7 +8,10 @@
             <group id="secondary" position="inside">
                 <group string="Inventory" groups="stock.group_stock_user">
-                    <field name="product_id" />
+                    <field 
+                        name="product_id"
+                        domain="[('maintenance_ok', '=', true)]"
+                     />
```
```suggestion
                    <field name="product_id" domain="[('type', '=', 'product')]" />
```

marcos-mendez committed:
Update fsm_equipment.xml

Correcting what Max asked to this to work. I really don't know if im doing the right thing, just trying again

max3903 commented:
@marcos-mendez Click on "Commit suggestion", that should fix the errors...

max3903 commented:
@marcos-mendez Couple steps that should help you:

```
pip install pre-commit
pre-commit install
pre-commit run -a
```

marcos-mendez commented:
I had never been able to install pre-commit with l10n-brazil installations because of this. 
![image](https://user-images.githubusercontent.com/38361760/206577040-0505dff2-4b8a-4464-acd8-47a0715dfeea.png)

max3903 commented:
@marcos-mendez Try with:

```
pip install --user pre-commit
```
or install `pipx` and use it to install pre-commit:

```
pipx install pre-commit
```

marcos-mendez commented:
Ok it worked. But now 
![image](https://user-images.githubusercontent.com/38361760/206776697-01a9a40b-768d-4d8a-8f07-c45c66eda097.png)

marcos-mendez commented:
Maybe i shloud close this and keep only the #1016

marcos-mendez commented:
I'll open it later whe i figure it out how to make a commit only to this PR
